src/Forms/Shared/Samples/Data/AddFeaturesWithContingentValues/AddFeaturesWithContingentValues.xaml.cs

```diff
@@ -0,0 +1,433 @@
+            {
+                return true;
+            }
+            else
+            {
+                foreach (ContingencyConstraintViolation violation in contingencyConstraintViolations)
+                {
+                    fieldGroupNames.Add(violation.FieldGroup.Name);
+                }
+            }
+
+            return false;
+        }
+
+        private void CreateNewNest()
+        {
+            // Once the attribute map is filled and validated, save the feature to the geodatabase feature table.
+            _ = _geodatabaseFeatureTable.UpdateFeatureAsync(_newFeature);
+
+            _ = QueryAndBufferFeatures();
+
+            _newFeature = null;
+        }
+
+        private async Task DiscardFeature()
+        {
+            try
+            {
+                // Delete the newly created feature from the geodatabase feature table.
+                await _geodatabaseFeatureTable.DeleteFeatureAsync(_newFeature);
+
+                _newFeature = null;
+            }
+            catch (Exception e)
+            {
+                await Application.Current.MainPage.DisplayAlert("Error", e.Message, "OK");
+            }
+        }
+
+        private async Task UpdateField(string field, object value)
+        {
+            try
+            {
+                // Update the feature's appropriate attribute based on a given field name and a value.
+                switch (field)
+                {
+                    case "Status":
+                        _newFeature.SetAttributeValue(field, _statusValues[value.ToString()]);
+                        break;
+
+                    case "Protection":
+                        _newFeature.SetAttributeValue(field, _protectionValues[value.ToString()]);
+                        break;
+
+                    case "BufferSize":
+                        _newFeature.SetAttributeValue(field, Convert.ToInt32(value));
+                        break;
+
+                    default:
+                        await Application.Current.MainPage.DisplayAlert("Error", $"{field} not found in any of the data dictionaries.", "OK");
+                        break;
+                }
+            }
+            catch (Exception e)
+            {
+                await Application.Current.MainPage.DisplayAlert("Error", e.Message, "OK");
+            }
+        }
+
+        private async Task SetProtectionComboItems()
+        {
+            List<string> protectionItems = await GetContingentValues("Protection", "ProtectionFieldGroup");
+
+            // Get the appropriate contingent values for populating the protection combo component.
+            if (protectionItems.Any())
+            {
+                ProtectionPicker.SelectedIndexChanged -= ProtectionCombo_Selected;
+                ProtectionPicker.ItemsSource = protectionItems;
+                ProtectionPicker.SelectedIndexChanged += ProtectionCombo_Selected;
+            }
+        }
+
+        private async Task SetBufferSizeSliderRange()
```

Comment:
This isn't always called when it should be for Forms. I was easily able to get into a situation where Unoccupied had a minimum of 100 and couldn't be placed.